service/service.go

```diff
@@ -50,6 +50,8 @@ type Application struct {
 	builtPipelines  builder.BuiltPipelines
 	builtExtensions builder.Extensions
 
+	exportersMap map[configmodels.DataType]map[configmodels.Exporter]component.Exporter
```

Comment:
May consider same optimization for extensionsMap (we construct it every time is called). We can cache that result as well.

Reply:
This is going to be called rarely so performance impact is likely negligible. We can add a cache if we see it is necessary in the future.

Reply:
I was more calling for consistency :)

Reply:
Yes, you are right. Now it is consistent.